src/map.rs

```diff
@@ -1711,13 +1697,11 @@ where
             // we are in one of three cases:
             //
             //  1. if table is the one we read before the loop, then we read it while holding the
-            //     guard, so it won't be dropped until after we drop that guard b/c the drop logic
-            //     only queues a drop for the next epoch after removing the table.
+            //     guard, so it won't be dropped until after we drop that guard b/c this thread must
+            //     have been included in the reference count of a retirement.
             //
-            //  2. if table is read by init_table, then either we did a load, and the argument is
-            //     as for point 1. or, we allocated a table, in which case the earliest it can be
-            //     deallocated is in the next epoch. we are holding up the epoch by holding the
-            //     guard, so this deref is safe.
+            //  2. if table is read by init_table, we did so while holding a guard, so the
+            //     argument is as for point 1.
```

Comment:
This is missing the "or" part of the argument now.

Reply:
Fixed in cc93a0c01bf622dae4280130c4999b135ba4d05e.